### PR TITLE
MDN JavaScript

### PR DESCRIPTION
**Pull request by bfmags**
Creating a new Longtail IA for MDNJS based on existing Fathead IA, to improve it's triggering and cover more queries.

https://github.com/duckduckgo/zeroclickinfo-fathead/issues/239

---

IA Page: https://duck.co/ia/view/mdn_js

### CODE REVIEW

**Comment by daxtheduck** [comment]
## [Mdn Javascript](https://duck.co/ia/view/mdn_js)

**Description**: Method signatures for core JavaScript 1.5 from the Mozilla Developer Network

**Example Query**: [js split](https://beta.duckduckgo.com/?q=js split), [javascript array](https://beta.duckduckgo.com/?q=javascript array)

**Tab Name**: About

**Source**:  

*These are the important fields from the IA page.  Please check these for errors or missing information and update the [IA page](https://duck.co/ia/view/mdn_js)*

---
*This is an automated message which will be updated as changes are made to the [IA page](https://duck.co/ia/view/mdn_js)*

File: lib/longtail/mdnjs/toxml.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+XML = """<?xml version="1.0" encoding="UTF-8"?>
+<add allowDups="true">
+{results}
+</add>
+"""
+
+DOC = """<doc>
+	<field name="title"><![CDATA[{title}]]></field>
+	<field name="paragraph"><![CDATA[{abstract} <a href="{source_url}">{source_url}</a>]]></field>
```
Comment (jdorweiler) [inline lib/longtail/mdnjs/toxml.py]: Can you add a `l2_sec` field too. 
```
<field name="title"><![CDATA[{title_strip}]]></field>
```

Then on line 28 make a copy of the title field with punctuation removed. Feel free to implement that however you want. This is just how I did it. 
```
line['title_strip'] = re.sub(r'[^\w\s]',' ',line['title'])
```

Comment (bfmags) [inline lib/longtail/mdnjs/toxml.py]: Sure can. I had l2_sec initially with the abstract but without links. What content would you place in the l2_sec? Thnx for your feedback

Comment (MariagraziaAlastra) [inline lib/longtail/mdnjs/toxml.py]: The l2_sec would have the title, but without punctuation (that'd be replaced by spaces)

Comment (bfmags) [inline lib/longtail/mdnjs/toxml.py]: cool! thnx guys

Comment (bfmags) [inline lib/longtail/mdnjs/toxml.py]: commited changes adding l2_sec field to xml and json output

https://github.com/bfmags/zeroclickinfo-longtail/commit/4264b461cfc223e648859d7cffdb36b261d76ae1

**Pull request closed without merging**